### PR TITLE
BEAM-642 - Support Flink Detached Mode for JOB execution

### PR DESCRIPTION
**Pull request by sumitchawla**
FlinkRunner gives exception when job is submitted in detached mode.
Following code will throw exception:
```
 LOG.info("Execution finished in {} msecs", result.getNetRuntime()); 
```

### CODE REVIEW

**Comment by aljoscha** [comment]
@mxm you probably know best what's going on here, could you maybe review?

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java [hunk 1]
```diff
@@ -151,18 +153,22 @@ public FlinkRunnerResult run(Pipeline pipeline) {
...
     }
 
-    LOG.info("Execution finished in {} msecs", result.getNetRuntime());
-
-    Map<String, Object> accumulators = result.getAllAccumulatorResults();
-    if (accumulators != null && !accumulators.isEmpty()) {
-      LOG.info("Final aggregator values:");
+    if (result instanceof DetachedEnvironment.DetachedJobExecutionResult) {
+      LOG.info("Pipeline submitted in Detached mode");
+      return new FlinkRunnerResult(new HashMap<String, Object>(), 0L);
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java]: I would pass `Collections.emptyMap()` and `-1` as runtime.

Comment (sumitchawla) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java]: @mxm Thanks for the review. I have updated the code as suggested by you.

**Comment by sumitchawla** [comment]
can this merge request be merged now? I need a build which supports this change.

**Comment by mxm** [comment]
Yes, it is ready to be merged.

**Pull request closed without merging**